webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,389 @@
+              console.log(owns.ids[0]);
+              console.log(v.attrs.name);
+
+              owns.ids.length > 0
+                ? rs.rsJsonApiRequest(
+                    "/rsIdentity/createIdentity",
+                    {
+                      id: owns.ids[0],
+                      name: v.attrs.name,
+                      pseudonimous: false,
+                      pgpPassword: passphase,
+                    },
+                    (data) => {
+                      const message = data.retval
+                        ? "Successfully created identity."
+                        : "An error occured while creating identity.";
+                      console.log(message);
+                      widget.popupMessage([
+                        m("h3", "Create new Identity"),
+                        m("hr"),
+                        message,
+                      ]);
+                    }
+                  )
+                : widget.popupMessage([
+                    m("h3", "Create new Identity"),
+                    m("hr"),
+                    "An error occured while creating identity.",
+                  ]);
+            });
+          },
+        },
+        "Enter"
+      ),
+    ],
+  };
+};
 const CreateIdentity = () => {
   // TODO: set user avatar
-  let name = '',
-    pgpPassword = '',
+  let name = "",
+    id = "",
+    pgpPassword = "",
     pseudonimous = false;
   return {
-    view: () => [
-      m('i.fas.fa-user-plus'),
-      m('h3', 'Create new Identity'),
-      m('hr'),
-      m('input[type=text][placeholder=Name]', {
+    view: (v) => [
+      m("i.fas.fa-user-plus"),
+      m("h3", "Create new Identity"),
+      m("hr"),
+      m("input[type=text][placeholder=Name]", {
         value: name,
-        oninput: e => (name = e.target.value),
+        oninput: (e) => (name = e.target.value),
       }),
-      'Type:',
       m(
-        'select',
+        "div",
         {
-          value: pseudonimous,
-          oninput: e => (type = e.target.value),
+          style: "display:inline; margin-left:5px;",
         },
         [
-          m('option[value=false]', 'Linked to your Profile'),
-          m('option[value=true]', 'Pseudonymous'),
-        ],
+          "Type:",
+          m(
+            "select",
+            {
+              value: pseudonimous,
+              style: "border:1px solid black",
+              oninput: (e) => {
+                pseudonimous = e.target.value === "true";
+                console.log(pseudonimous);
+              },
+            },
+            [
+              m("option[value=false][selected]", "Linked to your Profile"),
+              m("option[value=true]", "Pseudonymous"),
+            ]
+          ),
+        ]
       ),
-      m('br'),
-      m('input[type=password][placeholder=Retroshare Passphrase]', {
+      m("br"),
+      m("input[type=password][placeholder=Retroshare Passphrase]", {
```

Comment:
Sorry this is confusing to me. Why do we have the passphrase field in two places? In lines 104 and 15?

Reply:
Actually, for creating a signed Identity, we need to pass a passphrase password but for an unsigned identity, we do not have to pass a passphrase password.
In line 15, that password is used as a login password for the Unsigned Identity.

Reply:
One more I observed, we do not even need to pass the login password for creating an unsigned identity. If you want then I will remove the line 15

Reply:
Yup would be great if it was removed.
Does not make sense to have it IMO.

Reply:
Sure @rottencandy.

Reply:
@rottencandy, I have updated the Changes. please review it.